query_table_create.go

```diff
@@ -227,8 +229,21 @@ func (q *CreateTableQuery) appendUniqueConstraints(fmter schema.Formatter, b []b
 		keys = append(keys, key)
 	}
 	sort.Strings(keys)
-
 	for _, key := range keys {
+		if q.table.SoftDeleteField != nil && q.Dialect().Features().Has(feature.PartialIndex)  {
+			keyName := fmt.Sprintf("%s_%s_unique_key", q.table.Name, key)
+			index := q.NewCreateIndex().
+				Index(keyName).
```

Comment:
We somehow should chose naming schema for these indexes so we can check the presence of the index by name in the hypothetical `AutoMigrate` method.